### DISCUSSION TITLE
Performance of binding the same ressource multiple times redundantly

### DISCUSSION THREAD

**Discussion by AndriBaal** [gfx-rs/wgpu#4085] [Q&A] [answered]
Does wgpu call the underlying API methods twice when binding the same pipeline, index buffer, vertex buffer or bindgroup on the same slot twice in a row? Should I check before binding the same Model again, to prevent unnecessary binding, or does wgpu do this by itself?

**Comment by cwfitzgerald** [comment, marked as answer]
There are limited situations where we do this, but assume everything you call will be executed on.
